cirq-google/cirq_google/engine/engine.py

```diff
@@ -149,6 +153,7 @@ def __init__(
         verbose: Optional[bool] = None,
         timeout: Optional[int] = None,
         context: Optional[EngineContext] = None,
+        serializer: Serializer = CIRCUIT_SERIALIZER,
```

Comment:
In the future we'd like to remove `SerializableGateset`, making `CIRCUIT_SERIALIZER` the only serializer AFAIK. What if we don't provide an option to configure the serializer and always use CIRCUIT_SERIALIZER?

Reply:
We could certainly do that, though having it configurable gives us a way to test out `CircuitSerializer2` when that comes out...

(I kid, but only slightly)

Reply:
I would lean towards keeping the code simpler now and adding this back when the use case comes up.

Another reason to keep the parameter is if we want to inject a mock to test that Engine calls the serializer with the expected circuit, but it doesn't look like we do that today. And I'm not sure if we should - if the logic for prepping the serializer call isn't complicated then such a test is probably overkill.

Reply:
Removed the option to customize this on `Engine`. I've kept it on `EngineContext` since that gets passed around to various places where needed and I'd prefer to access the serializer in one way on the context instead of hard-coding `CIRCUIT_SERIALIZER` everywhere it is used.